Braintree/Payments/@Public/BTPaymentProvider.h

```diff
@@ -18,6 +18,29 @@ typedef NS_ENUM(NSInteger, BTPaymentProviderType) {
     BTPaymentProviderTypeApplePay
 };
 
+/// Status of the last (or ongoing) payment method creation
+typedef NS_ENUM(NSInteger, BTPaymentProviderStatus) {
+    /// The payment method was not created yet (default status)
+    BTPaymentProviderStatusUninitialized = 0,
+
+    /// Payment method created (either PayPal, Venmo or ApplePay)
+    BTPaymentProviderStatusInitialized,
+
+    /// The payment method creator, having obtained user payment details and/or user
+    /// authorization, is now processing the results
+    BTPaymentProviderStatusProcessing,
+
+    /// Payment method failed with error
+    BTPaymentProviderStatusError,
+
+    /// Payment method creation is complete with success
+    BTPaymentProviderStatusSuccess,
+
+    /// The payment method creator has cancelled
+    BTPaymentProviderStatusCancelled
```

Comment:
Super minor, but we apparently favor "Canceled" with one 'l' in our codebase. e.g.

https://github.com/braintree/braintree_ios/blob/a1522fecf8bd14f50919594cb7b8d20e8a827d9f/Braintree/Venmo/BTVenmoAppSwitchReturnURL.h#L9

Apparently this is an American vs British thing.
http://www.quickanddirtytips.com/education/grammar/canceled-or-cancelled